plugins/tiddlywiki/webdav/webdavadaptor.js

```diff
@@ -0,0 +1,248 @@
+						// var type = prop.getElementsByTagNameNS(ns, 'getcontenttype')[0].innerHTML;
+						if (href.endsWith(".tid") && href!=="$__StoryList.tid") {
+							var type = "application/x-tiddler";
+							// var revision	 = prop.getElementsByTagNameNS(ns, 'getetag')[0].innerHTML;
+							var revision	 = prop.getElementsByTagNameNS(ns, 'getlastmodified')[0].innerHTML;
+						  href = href.slice(0, -4);
+						  tiddlers.push({ title: href,
+										revision: revision,
+										type: type });
+						}
+					}
+					if (collection) {
+						if (!(self.knownDirs.includes(href))) {
+							self.knownDirs.push(href);
+						}
+					}
+				}
+				callback(null, tiddlers);
+			}
+		}
+	});
+};
+
+/*
+Save a tiddler and invoke the callback with (err,adaptorInfo,revision)
+*/
+WebDAVAdaptor.prototype.saveTiddler = function(tiddler,callback) {
+	var self = this;
+	// Save the tiddler as a self contained templated file
+	var content = self.wiki.renderTiddler("text/plain","$:/core/templates/tid-tiddler",
+		{variables: {currentTiddler: tiddler.fields.title}});
+	var url = this.tiddlersBaseURL + "/" + encodeURIComponent(tiddler.fields.title) + ".tid";
+	var path = dirname(tiddler.fields.title);
+	if (path!==tiddler.fields.title && !(self.knownDirs.includes(path))) {
+		this.mkdir_p(path, callback);
+	}
+	$tw.utils.httpRequest({
+		url: this.tiddlersBaseURL + "/" + tiddler.fields.title + ".tid",
+		type: "PUT",
+		headers: {
+			"Content-type": "text/plain"
+		},
+		data: content,
+		callback: function(err,data,request) {
+			if(err) {
+				return callback(err);
+			}
+			// Save the details of the new revision of the tiddler
+			var revision = new Date().toGMTString(); // request.getResponseHeader("Etag");
+
+			// Invoke the callback
+			callback(null, {}, revision);
+		}
+	});
+};
+
+/*
+Load a tiddler and invoke the callback with (err,tiddlerFields)
+*/
+WebDAVAdaptor.prototype.loadTiddler = function(title,callback) {
+	var self = this;
+	$tw.utils.httpRequest({
+		url: this.tiddlersBaseURL + "/" + title + ".tid",
+		headers: {accept: "charset=UTF-8"},
+		callback: function(err,data,request) {
+			if(err) {
+				return callback(err);
+			}
+			// Invoke the callback
+			var type = "application/x-tiddler";
+			var tiddler = $tw.wiki.deserializeTiddlers(type, data)[0];
+			tiddler.revision = request.getResponseHeader("Last-Modified")
+			callback(null, tiddler);
+		}
+	});
+};
+
+/*
+Delete a tiddler and invoke the callback with (err)
+options include:
+tiddlerInfo: the syncer's tiddlerInfo for this tiddler
+*/
+WebDAVAdaptor.prototype.deleteTiddler = function(title,callback,options) {
+	// Issue HTTP request to delete the tiddler
+	$tw.utils.httpRequest({
+		url: this.tiddlersBaseURL + "/" + title + ".tid",
+		type: "DELETE",
+		callback: function(err,data,request) {
+			if(err) {
+				return callback(err);
+			}
+			// Invoke the callback
+			callback(null);
+		}
+	});
+};
+
+if($tw.browser && document.location.protocol.substr(0,4) === "http" ) {
```

Comment:
hmmm, I personally consider unencrypted webdav write access, to be extreamly dangerous. ... 

Our users will use the stuff and expose it to the web. Many of them don't ask, if it save to do so, they expect it to be save. Which it isn't.

With IIS its simple to create a "self-signed" cert even for local testing. And I think, we should force it.

Reply:
I think it should be the user's responsibility! We could add a note.

Reply:
I tend to disagree a little bit here. ... We need a ---- big note ----